src/depot_gui.cpp

```diff
@@ -916,6 +918,52 @@ struct DepotWindow : Window {
 		return true;
 	}
 
+	/**
+	 * Clones a vehicle from a vehicle list.
+	 * @return Returns true if clone/copy succeeded, false otherwise.
+	 */
+	bool OnVehicleSelect(VehicleList::const_iterator begin, VehicleList::const_iterator end) override
+	{
+		if (!_ctrl_pressed) {
+			/* If CTRL is not pressed: If all the vehicles in this list have the same orders, then copy orders */
+			if (AllEqual(begin, end, [](const Vehicle *v1, const Vehicle *v2) {
+				return VehiclesHaveSameEngineList(v1, v2);
+			})) {
+				if (AllEqual(begin, end, [](const Vehicle *v1, const Vehicle *v2) {
+					return VehiclesHaveSameOrderList(v1, v2);
+				})) {
+					OnVehicleSelect(*begin);
+				}
+				else {
+					ShowErrorMessage(STR_ERROR_CAN_T_BUY_TRAIN + (*begin)->type, STR_ERROR_CAN_T_COPY_ORDER_VEHICLE_LIST, WL_INFO);
+				}
+			}
+			else {
+				ShowErrorMessage(STR_ERROR_CAN_T_BUY_TRAIN + (*begin)->type, STR_ERROR_CAN_T_CLONE_VEHICLE_LIST, WL_INFO);
```

Comment:
Nope that would be wrong.  OnVehicleSelect() returns true to tell the caller that the mouse click is handled by the active contextual action, and so the click shouldn't be re-handled by the default action.  In this case, the active contextual action is to clone/copy a vehicle (because the "Clone Vehicle" button was pressed), and we shouldn't do the default action (e.g. open the vehicle window) even if the contextual action fails to be performed.  Showing an error message (and not doing anything else) is the correct and expected behaviour here.

In the part of my PR in group_gui.cpp and vehicle_gui.cpp you will see some code that looks like this:
```
if (!VehicleClicked(vehgroup)) {
    // <-- do the default action here (i.e. open either the vehicle view window or vehicle list window)
}
```